### PR TITLE
Add browser polyfills for webpack compilation.

### PR DESCRIPTION
**Pull request by jasongrout**
### Description

This adds the necessary polyfills, according to the spec at https://github.com/defunctzombie/package-browser-field-spec to enable webpack to ignore any imports of these modules when packing files for the browser.

To be clear, the "polyfills" added here are to just ignore the imports, since the imports shouldn't happen in the browser.

With this change, we can successfully webpack a simple js file that imports pyodide:
```js
import { loadPyodide } from "pyodide";
```
### Checklists

- [X] Add a [CHANGELOG](https://github.com/pyodide/pyodide/blob/main/docs/project/changelog.md) entry

### CODE REVIEW

**Review by rth — COMMENTED** [review]
That's great, thanks a lot for doing this @jasongrout !

Can we then remove the webpack related comments in https://github.com/pyodide/pyodide/blob/fda600e519414109b4b1ebb93b09723e08e29dea/src/js/compat.ts#L32 (and similar lines in that file)?

**Comment by jasongrout** [comment]
> Can we then remove the webpack related comments in

Great question! Yes, I think so. I'll try it.

**Comment by rth** [comment]
Hmm, looks like the webpack example in CI fails,
```
ERROR in ./node_modules/pyodide/load-pyodide.js 74:20-32
Module not found: Error: Can't resolve 'vm' in '/root/repo/pyodide-webpack-example/node_modules/pyodide'
```

File: .circleci/config.yml [hunk 1]
```diff
@@ -232,14 +232,14 @@ jobs:
...
-      #       npm ci
-      #       cp ../src/js/*.js node_modules/pyodide/
-      #       npx webpack
+      - run:
+          name: check if webpack cli works well with load-pyodide.js
+          command: |
+            git clone https://github.com/pyodide/pyodide-webpack-example.git
+            cd pyodide-webpack-example
+            npm ci
+            cp ../src/js/*.js node_modules/pyodide/
```
Comment (rth) [inline .circleci/config.yml:241]: So I think the issue was that that package specifies pyodide 0.18.0 https://github.com/pyodide/pyodide-webpack-example/blob/main/package.json#L18
while we want to use it with latest pyodide. So this was a workaround for that, but after migrating to typescript, I don't think that's correct anymore?

**Comment by rth** [comment]
It's also possible to merge this without running that test, and try to fix the test in a separate PR, as you prefer.

**Comment by jasongrout** [comment]
> It's also possible to merge this without running that test, and try to fix the test in a separate PR, as you prefer.

I went ahead and reverted back to having that test commented out, then.

**Pull request merged**